src/rate-limiter.ts

```diff
@@ -26,26 +24,27 @@ export class RateLimiter {
         return new Date().getTime() < retryAfter
     }
 
-    on429Response(response: XMLHttpRequest): void {
-        if (response.status !== 429) {
+    // this needs to be an arrow function so that it can be passed as a callback
+    // and have the correct `this` context in order to read `checkThreshold`
+    public checkForLimiting = (xmlHttpRequest: XMLHttpRequest): void => {
+        if (Math.random() >= this.checkThreshold) {
+            // we don't need to check this on every request
```

Comment:
I find this a little confusing. Why do we only need to check now and then? 
Feels like a step that could be confusing to us in the future when we still see a lot of requests coming in.

Is the subsequent code super slow or something? Seems like fairly straightforward code so I wouldn't say the perf benefits outweigh the extra complexity personally